src/VkLayer_GF_amber_scoop/include/VkLayer_GF_amber_scoop/amber_scoop_layer.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The gf-layers Project Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VKLAYER_GF_AMBER_SCOOP_AMBER_SCOOP_LAYER_H
+#define VKLAYER_GF_AMBER_SCOOP_AMBER_SCOOP_LAYER_H
+
+#include <vulkan/vulkan.h>
+
+#include <atomic>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "VkLayer_GF_amber_scoop/vulkan_commands.h"
+#include "gf_layers_layer_util/util.h"
+
+namespace gf_layers::amber_scoop_layer {
+
+// This class is used to keep track of a command buffer. This class stores
+// commands recorded to a command buffer and keeps track when the command buffer
+// has been submitted. Also keeps track if the command buffer being tracked
+// contains any draw calls. The class is not thread-safe, but it shouldn't
+// matter because the Vulkan spec requires Vulkan applications to record
+// commands without races.
+class CommandBufferTracker {
+ public:
+  // Add command to the command list.
```

Comment:
Function documentation comments must start with descriptive verbs in the indicative mood. E.g. "Adds |cmd|..." instead of "Add...". But then comments in function bodies are usually the other way around! I will change these.